src/oca_pre_commit_hooks/global_parser.py

```diff
@@ -0,0 +1,70 @@
+import argparse
+import configparser
+from os import getcwd
+from os.path import exists, expanduser, join
+
+CONFIG_NAME = ".oca_hooks.cfg"
+MSG_CTRL = "MESSAGES_CONTROL"
+
+
+def parse_disable(value):
+    return set(value.split(","))
+
+
+class GlobalParser(argparse.ArgumentParser):
+    def __init__(self):
+        super().__init__()
+        self.add_argument(
+            "--no-verbose",
+            action="store_true",
+            default=False,
+            help="If enabled so disable verbose mode.",
+        )
+        self.add_argument(
+            "--no-exit",
+            action="store_true",
+            default=False,
+            help="If enabled so it will not call exit.",
+        )
+        self.add_argument(
+            "--disable",
+            "-d",
+            type=parse_disable,
+            default=set(),
+            help="Disable the checker with the given 'check-name', separated by commas.",
+        )
+        self.add_argument(
+            "--enable",
+            "-e",
+            type=parse_disable,
+            default=set(),
+            help=(
+                "Enable the checker with the given 'check-name', separated by commas. "
+                "Default: All checks are enabled by default"
+            ),
+        )
+        self.add_argument("--config", "-c", type=str, help="Path to a configuration file")
+
+    def parse_args(self, args=None, namespace=None):
+        res = super().parse_args(args)
+
+        if not res.config and exists(join(getcwd(), CONFIG_NAME)):
+            res.config = join(getcwd(), CONFIG_NAME)
+        elif exists(join(expanduser("~"), CONFIG_NAME)):
+            res.config = join(expanduser("~"), CONFIG_NAME)
+
+        if res.config and not exists(res.config):
+            raise FileNotFoundError
+
+        if res.config:
+            conf = configparser.ConfigParser()
+            conf.read(res.config)
+
+            if conf.has_section(MSG_CTRL):
+                message_conf = conf[MSG_CTRL]
+                res.enable |= parse_disable(message_conf.get("enable", ""))
```

Comment:
IMHO the precedence of parameters should be:

1) command --parameter
2) configuration parameter
3) os.environ[parameter]
4) parameter default

Reply:
I mean, don't concatenate but replace

It is similar to pylint and odoo way

Reply:
If I understand correctly, if someone passes an `--enable` argument, we should discard whatever `enable` is in the cfg file, since it takes precedence?

Reply:
Yes

You can double-confirm it using `pylint` directly

We should use the same way